### DISCUSSION TITLE
korean account and character name

### DISCUSSION THREAD

**Discussion by aa1234saf** [evennia/evennia#3283] [Evennia Questions] [answered]
I don't speak English very well, so I write with a translator. I want to make this in Korean. But as soon as I start, I have a problem. I like the classic style and I don't want to separate the account and the character. So I set MULTISESSION_MODE = 0, AUTO_CREATE_CHARACTER_WITH_ACCOUNT = True. Then when I create account #1, there is no problem, the account and the character have the same name. But when I create a second normal account and enter the game, the character name is XXX. If I delete the character and enter the same character name as the account name with charcreate again, it works fine. So I created another account name in English and entered the game, and the character name was set as normal for the English account. Perhaps this problem occurs when the account is created and the character is created with a Korean account name, so where should I look?

**Comment by Griatch** [comment, marked as answer]
So to make sure I understand what's happening: 
- You create an Account with a Korean name (using Hangul letters I presume)
- When logging in under `MULTISESSION_MODE=0`, the Character name becomes `XXX`.

What you need to do is override the `mygame/typeclasses/characters.py` `Character class` e.g. like this: 

```py 
# in typeclasses/characters.py 

# ... 

class Character(ObjectParent, DefaultCharacter): 

    # ... 

    @classmethod
    def normalize_name(cls, name): 
        return name 
```
This is because by default, the `normalize_name` method is "latinifiying" the character name - this is done for security reasons, to avoid people putting Unicode characters in there that make it look like another name. Non-latin recognized letters are replaced by X, which is why you see `XXX` since Hangul is obviously not Latin-based. Just returning the incoming name should fix this (you could of course apply any specific Hangul-checks you'd want on the name here too). 

> (Side note; we should update this method to look at the `LANGUAGE_CODE` value before applying `latinify` to avoid users of non-latin based languages to have to face this kind of thing).

**Reply by aa1234saf** [reply]
Thank you, you solved the problem!
